test/functorch/test_control_flow.py

```diff
@@ -73,6 +74,49 @@ def f(x, pred, pred2):
 
         self.assertEqual(result_false_true, torch.cos(x))
 
+    def test_cond_functionalized(self):
+        def true_fn(x):
+            return x.sin().max()
+
+        def false_fn(x):
+            return x.cos().min()
```

Comment:
maybe add some (internal) mutations in each branch, so you can assert that the final graph doesn't have any mutations in it?